tests/unit/backend/test_tiledb.py

```diff
@@ -0,0 +1,273 @@
+        # create collection
+        id = self.create_collection()
+        # add dataset
+        did = self.db.add_dataset(id, self.dataset_metadata)
+        # edit dataset
+        self.db.edit_dataset(did, "cell_count", 13)
+        # read previous version
+        dataset = self.db.read_dataset_history(did, 1)  # 1 version back
+        self.assertEqual(dataset['cell_count'], 12)
+        # revert to previous version
+        self.db.revert_dataset_history(did, 1)
+        dataset = self.db.get_dataset(did)
+        self.assertEqual(dataset['cell_count'], 12)
+
+    def test_create_revision_on_collection(self):
+        # Create a collection
+        id = self.create_collection()
+        # Create the revision
+        rev_id = self.db.create_revision(id)
+        # Get the revision
+        rev = self.db.get_collection(rev_id)
+        self.assertEqual(rev['name'], "test_collection")
+
+    def test_edit_revision(self):
+        # Create a collection
+        id = self.create_collection()
+        # Create a revision
+        rev_id = self.db.create_revision(id)
+        # Edit the revision
+        self.db.edit_collection(rev_id, "name", "edited_name")
+        # Get the revision
+        rev = self.db.get_collection(rev_id)
+        self.assertEqual(rev['name'], "edited_name")
+
+    def test_publish_revision(self):
+        # Create a collection
+        id = self.create_collection()
+        # Create a revision
+        rev_id = self.db.create_revision(id)
+        self.db.edit_collection(rev_id, "name", "edited_name")
+        # Publish the revision
+        self.db.publish_revision(rev_id)
+        # Get the new collection
+        coll = self.db.get_collection(id)
+        self.assertEqual(coll['name'], "edited_name")
+
+    def test_delete_revision(self):
+        # Create a collection
+        id = self.create_collection()
+        # Create a revision
+        rev_id = self.db.create_revision(id)
+        # Delete the revision
+        self.db.delete_collection(rev_id)
+        # Try getting the revision
+        status = self.db.get_attribute(rev_id, "visibility")
+        self.assertEqual(status, "DELETED")
+
+    def test_get_published_collections(self):
+        _ = self.create_collection()
+        id2 = self.create_collection()
+        self.db.publish_collection(id2)
+        colls = self.db.get_published_collections()
+        self.assertEqual(colls[0]['id'], id2)
+
+    def test_get_published_datasets(self):
+        id1 = self.create_collection()
+        _ = self.db.add_dataset(id1, self.dataset_metadata)
+        id2 = self.create_collection()
+        d2 = self.db.add_dataset(id2, self.dataset_metadata)
+        self.db.publish_collection(id2)
+        datasets = self.db.get_published_datasets()
+        self.assertEqual(datasets[0]['id'], d2)
+
+    def test_check_collection_access(self):
+        id = self.create_collection()
+        self.db.edit_collection(id, "visibility", "PRIVATE")
```

Comment:
This is a bit borderline. `edit_collection` should only be able to modify metadata, not an attribute that should only be managed internally. Maybe create a `_edit_collection` method that works on each attribute and expose an `edit_collection` that works on metadata only.